src/entities/Basket/Basket.interface.ts

```diff
@@ -0,0 +1,16 @@
+import * as mongoose from 'mongoose';
+
+type StatusType = 'pending'| 'accepted'|'rejected'|'cancelled'|'offered'|'failedByRequestor'|'failedByTarget'
+
+export interface Basket{
+    _id: mongoose.Types.ObjectId,
+    createdByUserId? : mongoose.Types.ObjectId,
+    targetUserID: mongoose.Types.ObjectId,
+    booksOffered: mongoose.Types.ObjectId[],
+    booksRequested: mongoose.Types.ObjectId[],
+    status: StatusType,
+    timeCreated: Date
```

Comment:
status and timeCreated also should be optional, as they will be assigned automatically